syncpack.config.cjs

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	indent: "\t",
+
+	// Don't set dep versions based on the version of the package in the workspace
+	workspace: false,
+
+	// Custom types are used to define additional fields in package.json that contain versions that should be
+	// checked/synced. See https://github.com/JamieMason/syncpack/blob/master/README.md#customtypes for more details.
+	customTypes: {
+		enginesNpm: {
+			path: "engines.npm",
+			strategy: "version",
+		},
+		enginesNode: {
+			path: "engines.node",
+			strategy: "version",
+		},
+		packageManager: {
+			path: "packageManager",
+			strategy: "name@version",
+		},
+	},
+
+	// semverGroup rules are applied in order to package/dep combinations. First matching rule applies. When running
+	// `syncpack lint-semver-ranges`, the output is grouped into numbered groups. However, the numbers of the groups are
+	// the _inverse_ of their order in this array. That is, Semver Group 1 in the output will correspond to the _last_
+	// rule in this array.
+	semverGroups: [
+		// GROUP 8
+		// engines.node should always use >= range
+		{
+			dependencyTypes: ["enginesNode"],
+			dependencies: ["**"],
+			packages: ["**"],
+			range: ">=",
+		},
+
+		// GROUP 7
+		// engines.npm should always use ^ range
+		{
+			dependencyTypes: ["enginesNpm"],
+			dependencies: ["**"],
+			packages: ["**"],
+			range: "^",
+		},
```

Comment:
@JamieMason This is the possible bug in custom types. I've defined `engines.node` and `engines.npm` custom types, but in testing I could only get one to trigger at a time. I can do more testing and give more details if it's helpful. I'm also happy to file a proper issue. Thanks for taking the time to help!

Reply:
This is probably a bug, I'll take a look 👍

Reply:
I tried to reproduce this [in this commit](https://github.com/JamieMason/syncpack/commit/97b3942c5666a1a5011c4a929512447c82b01c5c) but the tests I expected to fail are actually passing – could you create an issue with anything more you can think of? thanks!

Reply:
I did a bit more testing on this and found several inconsistencies. I wrote them up a readme on my branch along with how to reproduce them with our config. Details are at https://github.com/tylerbutler/FluidFramework/tree/syncpack-issues-repro

Reply:
hey @tylerbutler, I wasn't sure where best to reply but I've put some comments to what's in the readme [in this gist](https://gist.github.com/JamieMason/a042cc26e5432df39cfe5f72877b059d)

Reply:
@JamieMason Thanks for looking into those issues and resolving them so quickly! I definitely had some config bugs on my end so thanks for looking over them. I updated to 9.8.4 and verified that all the issues I noticed were fixed. And the new group headers are _extremely helpful_ for me! Cheers!